src/util/rational_gmp_imp.h

```diff
@@ -95,11 +95,11 @@ class Rational
   Rational(unsigned long int n) : d_value(n, 1) { d_value.canonicalize(); }
 
 #ifdef CVC5_NEED_INT64_T_OVERLOADS
-  Rational(int64_t n) : d_value(static_cast<long>(n), 1)
+  Rational(int64_t n) : d_value(static_cast<signed long int>(n), 1)
```

Comment:
Shouldn't these be long long, similar to the other changes?

Reply:
They got changed in the second commit ...

* GMP works on both Windows and Linux; and GMP *does not* have a `long long` overload (at least on Windows)
* CLN only works on Linux, and *does* have a `long long` overload

In actually gets a bit worse than this: if you make both CLN and GMP be `long int` then the separation logic API tests still fail on 32-bit Debian -- it seems it really has to be `(un)signed long long` to make CLN be happy on 32-bit Debian.

Basically, yes, I'd like them both to be `long long`, but `(un)signed long int` was the best I could get for GMP on Windows.

Reply:
Alright, got it. Can you document this in the header accordingly? (Just to avoid that later changes try to "fix" this).

Reply:
@mpreiner: okay, I implemented something, but something you might "hate": take a look at the most recent commit.

Reply:
The PR now takes a totally different approach, so these comments have been completely superseded.